core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -902,7 +909,12 @@ oppia.factory('explorationStatesService', [
         return getStatePropertyMemento(stateName, 'answer_groups');
       },
       saveInteractionAnswerGroups: function(stateName, newAnswerGroups) {
-        saveStateProperty(stateName, 'answer_groups', newAnswerGroups);
+        var newAnswerGroupsToBackend = [];
+        for (var i = 0; i < newAnswerGroups.length; i++) {
+          newAnswerGroupsToBackend.push(newAnswerGroups[i].toBackendDict());
+        }
+        saveStateProperty(stateName, 'answer_groups',
+          newAnswerGroups, newAnswerGroupsToBackend);
```

Comment:
Actually I suggest you keep the existing interface here and do the backend conversion within saveStateProperty. Would that work? It seems like a good idea to limit the backend representations to as small a part of the code as possible, and besides the current approach is basically passing duplicate args (since the fourth can be derived from the third).